### ISSUE TITLE
Java service client does not set correlationId

### ISSUE THREAD

**Issue by michaelbironneau** [Azure/azure-iot-sdks#911]
I'm using the following code to send a message via the service client, and then the block after that to receive the feedback:

```
Message signalMessage = new Message(signal.toString());
            signalMessage.setCorrelationId(signal.getSignalId().toString());
            signalMessage.setDeliveryAcknowledgement(DeliveryAcknowledgement.Full);
            serviceClient.send(signal.getDeviceId(), signalMessage);
```

Receiving feedback:

```
FeedbackBatch feedbackBatch;
feedbackBatch = receiver.receive(this.timeout);
if (feedbackBatch != null) {
                    List<FeedbackRecord> records = feedbackBatch.getRecords();
                    for (FeedbackRecord r : records) {
                        logger.info("Correlation Id: " + r.getCorrelationId() + " | Description: " + r.getDescription());
                }
```

Unfortunately the correlation Id is always empty. Is this expected behavior, and if so, how can I know which FeedbackRecord corresponds to which Message?

Thanks

**Comment by tameraw** [comment]
@michaelbironneau - Thanks for reporting this issue. Can you please provide which version of the SDK where you using?

**Comment by michaelbironneau** [comment]
@tameraw We are using 1.0.7.

**Comment by anthonyvercolano** [comment]
Hi,

 It is the designed behavior in this case for the correlationId to be blank.  I understand this might not be expected.  To determine the id of the corresponding message refer to the field originalMessageId.

Unfortunately the designed behavior was not implemented appropriately until release 1.0.10.

https://msdn.microsoft.com/en-us/library/microsoft.azure.devices.feedbackrecord.originalmessageid.aspx

**Comment by michaelbironneau** [comment]
Thanks, I can see how it changed in 1.0.10. Will upgrade our SDK dependency.
